### DISCUSSION TITLE
How to move nodes between multiple editors?

### DISCUSSION THREAD

**Discussion by yf-yang** [facebook/lexical#5206] [Q&A] [answered]
Scenario: drag and drop a node to a nested editor.

Possible (but failed) approaches:
1. Directly remove a node from previous editor then add it to the new one
    Why it fails: LexicalNode's constructor comes with a side effect. It registers itself at the activeEditor (which means a node must be created and mounted in the same editor.update call)
2. Reuse lexical/clipboard utilities:
    Why it does not succeed: I am not sure if I am calling it in a wrong way. I found that the utility function `$generateJSONFromSelectedNodes` needs a selection to generate the JSON, so I have to make a transient selection. It is wrong to make a node selection because it doesn't cover the node's children. I still do not know what is the correct way to make a RangeSelection with only the expected node being selected. Another problem is, `$generateJSONFromSelectedNodes` cannot even be called within a `editorState.read` call because it internally calls `clone`, I really don't know why.
3. Write a function that recursively calls `exportJSON`
    This would be the last available approach. Before doing so, I'd like to ask if there exists better approaches.

**Comment by yf-yang** [comment, marked as answer]
So far the best approach I can think of:
1. copy the `exportNodeToJSON` source code from packages/lexical/src/LexicalEditorState.ts
2. import `$parseSerializedNode`

**Reply by flakis05** [reply]
Do you have an example of how you did this?When did you call parseSerializedNode?
